crypto/signer.go

```diff
@@ -31,39 +31,39 @@ var sigpbHashLookup = map[crypto.Hash]sigpb.DigitallySigned_HashAlgorithm{
 // Signer is responsible for signing log-related data and producing the appropriate
 // application specific signature objects.
 type Signer struct {
-	hash   crypto.Hash
```

Comment:
NewSigner hardcodes the hash. Instead of changing it to have a Hash and a Signer parameter, I've made the members public.

This PR is the first step of a larger change, which is to consider the tree configuration when creating hashers/signers. After that we may elect to remove NewSigner completely.